### PR TITLE
Style chapter page

### LINKED ISSUE #55: Style chapter pages
Style the chapter pages, make the playable area and text a bit bigger etc

### PR DESCRIPTION
**Pull request by DavidMatalik**
Rearrangement of code: 
- Added a lot of stuff into chapterDOM in order to style elements properly - was hard to read at some point. Changed it then to Tudwall style. I hope that's ok?
- Removed container in chapterDOM which contained all draggables and dropzone, because positioning didn't work properly.

Styling discussion:
- Whatever comes to your mind. xD
- The header is maybe a bit weak.  I tried to make it bold, but then I find it a bit too strong. I like it more when it is rather weaker then bolder, what do you think?
- Hovering over dropzone with right image effect is ok, but doesn't excite me - maybe you have better ideas. Should there also be some effect when wrong image hovers over dropzone?
- Maybe font-size or positioning of story-text/question-text need to change. Depends on the real texts - would wait with adjusting until the real texts are loaded.

Open things: 
- Load chapter headings dynamically - at the moment always chapter 1
- Load background image for left bottom corner dynamically - at the moment always kite

Peek 2021-07-31 11-27

### CODE REVIEW

**Comment by emdevs** [comment]
It's cool to see a styled version of our game :D Good work!

I have a couple ideas of what I'd change though:

- I think if the story text is instead centered in the middle of the chapter it would be good. I think the position of the question text is okay, it would be better though to have it be of a different font, so that it's more separated from the story text.
- You mention that you think the effect of the correct item being dropped isn't exciting enough, maybe we could add a white glow around the drop zone as an extra effect?
- I'm not really sure if an image in the bottom left is necessary. In my opinion, it won't be clear that it's for decoration since it's going to be the same style as the droppable images.

**Comment by Tudwall** [comment]
> 
> 
> It's cool to see a styled version of our game :D Good work!
> 
> I have a couple ideas of what I'd change though:
> 
>     * You mention that you think the effect of the correct item being dropped isn't exciting enough, maybe we could add a white glow around the drop zone as an extra effect?


I think a white glow would look nice as well!

> 
>     * I'm not really sure if an image in the bottom left is necessary. In my opinion, it won't be clear that it's for decoration since it's going to be the same style as the droppable images.
> 
>     * I think if the story text is instead centered in the middle of the chapter it would be good. I think the position of the question text is okay, it would be better though to have it be of a different font, so that it's more separated from the story text.

I agree with those points too!

**Comment by DavidMatalik** [comment]
> I have a couple ideas of what I'd change though:

Thanks for input! Removed the background image - and made the draggables and dropZone bigger. The text might not be an issue anymore because of movement to chapter intro? 

The glow effect sounds cool. I think this could be an extra PR. Don't know how to do it spontaneously. 

Also an extra PR: Put "next" inside the next arrow. Also don't know spontaneously how to do it.

Peek 2021-08-02 14-58

**Review by Tudwall — APPROVED** [review]
Looks good! For the glow effect, I think a white box-shadow would do it, but you're right, it can be done in another PR

**Comment by Tudwall** [comment]
I think you should commit your package.json as well so svg-url-loader is in it, it would avoid us having to reinstall it manually everytime

**Comment by DavidMatalik** [comment]
> I think you should commit your package.json as well so svg-url-loader is in it, it would avoid us having to reinstall it manually everytime

Did you have an error because of svg-loader? svg-loader isn't in my package.json and still it works for me... But yes I have it locally installed in npm folder... strange xD Good hint

File: src/js/chapterDOM.js [hunk 1]
```diff
@@ -1,80 +1,96 @@
...
+    return dragContainer;
   };
 
   const setupChapterPage = () => {
-    section = document.createElement("section");
     const { homeButton, nextChapterButton } = createButtons();
+    
+    const title = document.createElement("H1");
+    title.classList.add("chapter-title");
+    title.textContent = "Chapter 1";
```
Comment (mdlugosz1) [inline src/js/chapterDOM.js]: We can dynamically set chapter number by using getChapterNumber method:
`title.textConent = 'Chapter ${chapterObj.getChapterNumber()}'`

**Review by mdlugosz1 — COMMENTED** [review]
It looks very nice! Only one minor change which i mentioned in previous comment.

> Also an extra PR: Put "next" inside the next arrow. Also don't know spontaneously how to do it.

Probably after & before CSS pseudo selectors would do the trick. I can take care of it if you want me to do it.

**Comment by DavidMatalik** [comment]
> * I'm not really sure if an image in the bottom left is necessary. In my opinion, it won't be clear that it's for decoration since it's going to be the same style as the droppable images.

Now that I removed the background image xD - I think it would be very nice on the intro chapter page  (where only the story text is displayd). This changed during the last rebase on main. What do you think?

File: src/js/chapterDOM.js (hunk 1 above)
Comment (DavidMatalik) [reply src/js/chapterDOM.js]: Thx! I didn't know that it would be so easy. =)

**Comment by DavidMatalik** [comment]
> Probably after & before CSS pseudo selectors would do the trick. I can take care of it if you want me to do it.

Nice - haven't worked with pseudo classes so far. I tried it and it works on e.g. section element, but doesn't seem to work on image elements:

_"Note: The pseudo-elements generated by ::before and ::after are contained by the element's formatting box, and thus don't apply to replaced elements such as , or to \<br> elements."_(https://developer.mozilla.org/en-US/docs/Web/CSS/::before)

With with putting a div around the image, it should be possible though. Yeah would be great if you could take care of it.

**Comment by mdlugosz1** [comment]
> Probably after & before CSS pseudo selectors would do the trick. I can take care of it if you want me to do it.
> 
> Nice - haven't worked with pseudo classes so far. I tried it and it works on e.g. section element, but doesn't seem to work on image elements:
> 
> "Note: The pseudo-elements generated by ::before and ::after are contained by the element's formatting box, and thus don't apply to replaced elements such as , or to <br> elements."(https://developer.mozilla.org/en-US/docs/Web/CSS/::before)
> 
> With with putting a div around the image, it should be possible though. Yeah would be great if you could take care of it.

Thanks for additional info about those selectors and img elements :) So i will take care of it after branch merge.

**Review by mdlugosz1 — APPROVED** [review]

**Review by emdevs — APPROVED** [review]

**Pull request merged**